docs/dev/truncate-with-tablets.md

```diff
@@ -0,0 +1,79 @@
+# Background
+
+The CQL TRUNCATE query is obviously data-modifying since it deletes user data, however,
+unlike DELETE queries, it is not based on a tombstone that carries a write-timestamp,
+with which we can determine for each cell, if it was logically written before
+or after the tombstone timestamp.
+
+Instead, truncate propagates in the cluster using an out-of-band RPC and is currently
+applied locally in each node using the local wall clock of each node (not synchronized
+with each other), typically after flushing the tables and taking a snapshot of them,
+if auto-snapshot is enabled.
+
+This presents a problem when tablets are enabled since there could on-going tablet
+transitions in the background, that the user has no control on, like tablet migration,
+with which a tablet might escape truncate by migrating between nodes just at the wrong
+time so it won’t be considered for truncate by any node.
+
+# Introduction
+
+This design proposal relates to [Make TRUNCATE TABLE safe with tablets #16411](https://github.com/scylladb/scylladb/issues/16411).
+
+Originally, the plan was to use a Truncate tombstone as per [Truncate must be based on a tombstone #11230](https://github.com/scylladb/scylladb/issues/11230).
```

Comment:
It's better not to refer to issues or historical plans here. It should be a self-contained document, and part of the series that implements it (though it's good to have it reviewed ahead of time).